workers/unity/Assets/Gdk/Core/Tests/Editmode/Systems/CleanReactiveComponentsSystemTests.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Linq;
+using Improbable.Gdk.TestUtils;
+using NUnit.Framework;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core.EditmodeTests.Systems
+{
+    [TestFixture]
+    public class CleanReactiveComponentsSystemTests
+    {
+        private UnityTestWorker worker;
+
+        [SetUp]
+        public void Setup()
+        {
+            // Need to use a different worker instance for every test because the system's world needs to be re-created.
+            worker = new UnityTestWorker("worker-id", new Vector3());
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            worker.Dispose();
+        }
+
+        [Test]
+        [TestCase(typeof(NewlyAddedSpatialOSEntity))]
```

Comment:
How do we keep this set of testcases in sync with all of the types that exhibit this behaviour?

Reply:
Good question... I'm unsure!

Reply:
There is a separate ticket to make the clean system find what it needs to remove dynamically. That should make this a lot easier.

Reply:
>Part of that ticket should be updating this test too.
@samiwh, please update the ticket so that is explicitly the case.